test/pkg/environment/aws/expectations.go

```diff
@@ -362,6 +366,41 @@ func (env *Environment) GetAMIBySSMPath(ssmPath string) string {
 	return *parameter.Parameter.Value
 }
 
+// GetSSMPath gets the SSM path for an AMI release specified by the family, k8s version, architecture, and release offset.
+// A release offset of 0 will get you the latest release, 1 the previous release, and so on.
+// Note: this is tighly coupled with the naming conventions for AL2 and AL2023. This is **NOT** compatible with Bottlerocket, Ubuntu, or Windows AMIs.
+func (env *Environment) GetSSMPath(amiFamily string, k8sVersion string, arch string, releaseOffset int) string {
+	GinkgoHelper()
+	Expect(amiFamily).To(BeElementOf([]string{
+		v1beta1.AMIFamilyAL2,
+		v1beta1.AMIFamilyAL2023,
+	}))
+	pathDetails, ok := lo.Find(amifamily.GetAMIFamily(&amiFamily, nil).DefaultAMIs(k8sVersion), func(out amifamily.DefaultAMIOutput) bool {
+		return out.Requirements.Compatible(scheduling.NewRequirements(scheduling.NewRequirement(v1.LabelArchStable, v1.NodeSelectorOpIn, arch))) == nil
+	})
+	Expect(ok).To(BeTrue())
+	basePath := func() string {
+		splitPath := strings.Split(pathDetails.Query, "/")
+		Expect(len(splitPath)).To(BeNumerically(">", 2))
+		return strings.Join(splitPath[0:len(splitPath)-2], "/")
+	}()
+
+	parameters := []string{}
+	Expect(env.SSMAPI.GetParametersByPathPages(&ssm.GetParametersByPathInput{
+		Path:      lo.ToPtr(basePath),
+	}, func(out *ssm.GetParametersByPathOutput, lastPage bool) bool {
+		parameters = append(parameters, lo.Map(out.Parameters, func(p *ssm.Parameter, _ int) string {
+			return lo.FromPtr(p.Name)
+		})...)
+		return lastPage
+	})).To(Succeed())
+
+	// Ensure there are enough published AMIs to satisfy the offset. Note that there are two SSM paths for the latest release.
```

Comment:
Is that because of the image path having an SSM parameter for recommended or latest?

Reply:
Having to encode that logic in here feels a bit awkward to me. Are there ways that we could workaround that that would be a little more clear to a reader than just adding two to the total number of the offset

Reply:
Yep, there's the path with the full release name and an alias called recommended. Seems reasonable to reject any path with recommended in the path name, should be a little more readable.